client/src/main/java/com/linecorp/decaton/client/DecatonClient.java

```diff
@@ -112,4 +124,13 @@
     static <T> DecatonClientBuilder<T> producing(String topic, Serializer<T> serializer) {
         return new DecatonClientBuilder<>(topic, serializer);
     }
+
+    @Builder
+    @Value
+    class TaskMetaData {
```

Comment:
I think we name it `Metadata` rather than `MetaData` conventionally.  So `TaskMetadata` would be better.